elastic/beats
# Add a disable_host option to Filebeat inputs

## What does this PR do?

This adds a configuration option `publisher_pipeline.disable_host` to disable the addition
of `host.name` in events. By default Filebeat adds `host.name` to all events and we want
to be able to disable this for data sources that do not originate on the host (like cloud logs).

Relates #13920

## Why is it important?

ECS host.* fields should be populated with details about the host on which the event happened, or from which the measurement was taken. For some data sources Filebeat does not know that actual host and it should not use its own host name in these events. This gives a way to turn this off.

## Checklist




## How to test this PR locally

Run Filebeat and verify that the output does not have host.name.
```
---
filebeat.inputs:
- paths: test.log
   publisher_pipeline.disable_host: true
output.console.pretty: true
```

## Use cases

This will be used by several modules:

- aws
- azure
- googlecloud
- misp
- o365
- okta
- panw
- suricata (depends on whether it monitors a network tap or local host traffic)
- zeek (depends on whether it monitors a network tap or local host traffic)
